_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,667 @@
+---
+layout: blog_post
+title: "JHipster Microservice Deployed to Azure with Cosmos DB"
+author: andrew-hughes
+by: contractor
+communities: [devops,security,java,javascript]
+description: ""
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness|conversion
+---
+#### JHipster Azure AKS Microservice
+
+In this tutorial, you're going to deploy a JHipster-based, reactive microservice to Azure Kubernetes Service (AKS). You'll use Azure's Cosmos DB as a persistent store for one of the services. For security, you'll use Okta as an OAuth and OIDC provider. You'll also securely encrypt all secrets in the project configuration files using Kubernetes secrets and `kubeseal`. This tutorial focuses on deploying an already generated project to Azure AKS. It does not go into great detail about generating the project. To see how the project was generated using JHipster, take a look at [this previous tutorial](https://developer.okta.com/blog/2021/01/20/reactive-java-microservices). 
+
+The project has a few different pieces:
+
+- JHipster registry: a Eureka server for service discovery and a Spring Cloud Config server for centralized configuration management
+- Gateway: public gateway application built with Vue
+- Store: simple demo store API using Azure's Cosmo DB API for MongoDB
+- Blog: simple demo blog API using a Neo4J database
+
+{% img blog/jhipster-k8s-azure/microservice-structure.png alt:"Microservice structure" width:"800" %}{: .center-image }
+
+This tutorial has a lot of different technologies in it. I've tried to make it as simple and as explicit as possible, but it's probably helpful to have some basic knowledge of Docker and Kubernetes before you start.
+
+If you're already familiar with all the tech in this tutorial, you can skip ahead to the prerequisites section. If not, I'm going to explain them a little before we move on.
+
+[JHipster](https://www.jhipster.tech/) is a development platform that streamlines the generation, development, and deployment of both monolithic and microservice applications. It supports a dizzying array of frontend (Angular, React, and Vue) and backend (Spring Boot, Micronaut, Quarkus, Node.js, and .NET) technologies. It's designed to be deployed using Docker and Kubernetes, and can easily deploy to all the major cloud platforms, such as AWS, Azure, Heroku, Cloud Foundry, Google Cloud Platform, and OpenShift. 
```

Comment:
Since the backend technologies mentioned include blueprints, we could really dizzy up the frontend and add Ionic, React Native, and Svelte!

Reply:
Since Pivotal shutdown run.pivotal.io, I'm not sure we should mention Cloud Foundry. I'm not sure there's anyone else that allows it for public use. https://tanzu.vmware.com/content/pivotal-web-services-blog/pivotal-web-services-end-of-availability-announcement-and-timeline